daemon/exec_darwin.go

```diff
@@ -0,0 +1,16 @@
+// +build darwin
+
+package daemon
+
+import (
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/daemon/exec"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func (daemon *Daemon) execSetPlatformOpt(c *container.Container, ec *exec.Config, p *specs.Process) error {
+	if c.OS == "darwin" {
+		p.User.Username = ec.User
```

Comment:
do we need c.OS == "darwin" if we specified build tag and filename build target ?

Reply:
> c.OS == "darwin" {

Reply:
Is there any other related code which does similar job ? Don't we have c.OS = Linux container in darwin environment?

Reply:
Check here: docker/docker/daemon/exec_windows.go

Reply:
windows container (not Linux container in Linux VM on Windows host) is pure windows while ours should run Linux container on darwin by emulation, so it may not make sense with the if statement.

If you're really sure that we need the if statement, let me know.

Reply:
okay, I am removing the entire check and returning nil.

Reply:
If it returns nil, it does the same when c.OS==Linux.   Isn't it the case that we don't have to take care ?

Why not just removing if line ?

Reply:
okay.